core/drivers/crypto/caam/acipher/caam_prime.c

```diff
@@ -0,0 +1,882 @@
+	0x3a, 0xb8, 0xa2, 0xc3, 0xa8, 0xb1, 0xfe, 0x6f, 0xdc, 0x83, 0xdb, 0x39,
+	0x0f, 0x74, 0xa8, 0x5e, 0x43, 0x9c, 0x7b, 0x4a, 0x78, 0x04, 0x87, 0x36,
+	0x3d, 0xfa, 0x27, 0x68, 0xd2, 0x20, 0x2e, 0x87, 0x42, 0xaf, 0x1f, 0x4e,
+	0x53, 0x05, 0x9c, 0x60, 0x11, 0xbc, 0x33, 0x7b, 0xca, 0xb1, 0xbc, 0x91,
+	0x16, 0x88, 0x45, 0x8a, 0x46, 0x0a, 0xbc, 0x72, 0x2f, 0x7c, 0x4e, 0x33,
+	0xc6, 0xd5, 0xa8, 0xa3, 0x8b, 0xb7, 0xe9, 0xdc, 0xcb, 0x2a, 0x63, 0x43,
+	0x31, 0xf3, 0xc8, 0x4d, 0xf5, 0x2f, 0x12, 0x0f, 0x83, 0x6e, 0x58, 0x2e,
+	0xea, 0xa4, 0xa0, 0x89, 0x90, 0x40, 0xca, 0x4a, 0x81, 0x39, 0x4a, 0xb6,
+	0xd8, 0xfd, 0x0e, 0xfd, 0xf4, 0xd3, 0xa0, 0x2c, 0xeb, 0xc9, 0x3e, 0x0c,
+	0x42, 0x64, 0xda, 0xbc, 0xd5, 0x28, 0xb6, 0x51, 0xb8, 0xcf, 0x34, 0x1b,
+	0x6f, 0x82, 0x36, 0xc7, 0x01, 0x04, 0xdc, 0x01, 0xfe, 0x32, 0x35, 0x2f,
+	0x33, 0x2a, 0x5e, 0x9f, 0x7b, 0xda, 0x1e, 0xbf, 0xf6, 0xa1, 0xbe, 0x3f,
+	0xca, 0x22, 0x13, 0x07, 0xde, 0xa0, 0x62, 0x41, 0xf7, 0xaa, 0x81, 0xc2,
+	0xc1, 0xfc, 0xbd, 0xde, 0xa2, 0xf7, 0xdc, 0x33, 0x18, 0x83, 0x8a, 0x2e,
+	0xaf, 0xf5, 0xf3, 0xb2, 0xd2, 0x4f, 0x4a, 0x76, 0x3f, 0xac, 0xb8, 0x82,
+	0xfd, 0xfe, 0x17, 0x0f, 0xd3, 0xb1, 0xf7, 0x80, 0xf9, 0xac, 0xce, 0x41,
+	0x79, 0x7f, 0x28, 0x05, 0xc2, 0x46, 0x78, 0x5e, 0x92, 0x95, 0x70, 0x23,
+	0x5f, 0xcf, 0x8f, 0x7b, 0xca, 0x3e, 0xa3, 0x3b, 0x4d, 0x7c, 0x60, 0xa5,
+	0xe6, 0x33, 0xe3, 0xe1
+};
+
+/*
+ * Speedups for prime searching
+ *
+ * These values are products of small primes.  Information about the product
+ * preceeds it.
+ *
+ * Per Handbook of Applied Cryptography, Menezes et al, 4.4.1, one can compute
+ * the percentage of non-primes weeded out by checking for small prime factors
+ * in the candidates.  In the table below, "highest prime" is used for B, and
+ * "%weeded" is the number of candidates which get through this
+ * sieve.  As you can see, even with relatively few primes, there are
+ * diminishing returns to using larger numbers of primes.
+ *
+ * Percentage weeded:  1 - 1.12/ln B
+ *
+ * These can be used to compute GCD(prime, smallprime) before the Miller
+ * Rabin; this will weed out those candidates with 'small' primes before doing
+ * the costly modular exponentation inside of Miller-Rabin.  (If the result is
+ * not one, then the candidate has as a factor at least one of the small primes
+ * in the product).
+ *
+ * So, where is the sweet spot for the size of the product versus the size of
+ * the candidate?  Does it depend upon the size of the PKHA multiplier?  Hunt
+ * time for primes takes a long time to actually compute, and what are the
+ * stats for percentage of candidates that might be weeded out?  If not many,
+ * then there is an extra computation.
+ */
+struct smallprime {
+	const size_t length;
+	const uint8_t *data;
+};
+
+static const struct smallprime smallprimes[] = {
+	/*     sizes     | #primes | highest prime | %weeded */
+	/*  bits / bytes |         |                         */
+	/*    64 / 8     |   15    |          53   |    72   */
```

Comment:
These are the properties of the smallprime products.

Reply:
That much I understood. :-)
With or without these extra comments it's just magic gibberish to me. A link to where you got this data from or a way to interpret it would be nice. As far as I'm concerned right now:
``` C
	/*     sizes     | #primes | highest prime | %weeded */
	/*  bits / bytes |         |                         */
	/*    64 / 8     |   15    |          53   |    72   */
```
doesn't tell me anything useful.

Reply:
For example, to test if a 8 bytes number is prime, we compute the GCD of this number and the product of all prime numbers from 0 to 53 (there are 15 prime numbers in this interval). It allows us to speed up the process to find if the number is prime. In the example, doing this discards 72% of prime candidates before doing the miller-rabin test, which costly.
The method is detailed here : http://cacr.uwaterloo.ca/hac/about/chap4.pdf
Values of prime products have been computed internally by the CAAM design team :)

Reply:
Fair enough, please add this as a comment in the code for future reference.

Reply:
Ok, I will add the fact that values has bee pre-computed by the CAAM design team.
The method is already specified in the comment above as well as the reference to the `Handbook of Applied Cryptography, Menezes et al, 4.4.1`